sdk/metric/internal/aggregate/float64.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aggregate // import "go.opentelemetry.io/otel/sdk/metric/internal/aggregate"
+
+import "math"
+
+const (
+	// significandWidth is the size of an IEEE 754 double-precision
+	// floating-point significand.
+	significandWidth = 52
+	// SignificandMask is the mask for the significand of an IEEE 754
+	// double-precision floating-point value: 0xFFFFFFFFFFFFF.
+	significandMask = 1<<significandWidth - 1
+	// exponentWidth is the size of an IEEE 754 double-precision
+	// floating-point exponent.
+	exponentWidth = 11
+	// exponentBias is the exponent bias specified for encoding
+	// the IEEE 754 double-precision floating point exponent: 1023.
+	exponentBias = 1<<(exponentWidth-1) - 1
+	// exponentMask are set to 1 for the bits of an IEEE 754
+	// floating point exponent: 0x7FF0000000000000.
+	exponentMask = ((1 << exponentWidth) - 1) << significandWidth
+)
+
+// getNormalBase2 extracts the normalized base-2 fractional exponent.
+// Unlike Frexp(), this returns k for the equation f x 2**k where f is
+// in the range [1, 2).  Note that this function is not called for
+// subnormal numbers.
+func getNormalBase2(value float64) int {
+	rawBits := math.Float64bits(value)
+	rawExponent := (int64(rawBits) & exponentMask) >> significandWidth
+	return int(rawExponent - exponentBias)
+}
```

Comment:
This doesn't handle exact power-of-two values nor subnormal values. Why not use the full algorithm described [here](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#scale-zero-extract-the-exponent)?

Reply:
Those are optimizations/corrections that are done when getting the bin Index, `getExpoBin()` and `getLogBin()`

Reply:
The problem isn't that this doesn't handle subnormals, it's that `Math.Log()` gives incorrect answers for subnormal numbers.
For example `math.Log(math.SmallestNonzeroFloat64)` should [yield -744.44007192138126231410729](https://www.wolframalpha.com/input?i2d=true&i=ln%5C%2840%29Power%5B2%2C%5C%2840%29-1022%5C%2841%29%5D*Power%5B2%2C%5C%2840%29-52%5C%2841%29%5D%5C%2841%29) but it [returns -709.0895657128241
](https://goplay.tools/snippet/ISoM5PFjk_b).  This means that supporting subnormals here would enable us to support subnormal numbers [here](https://github.com/open-telemetry/opentelemetry-go/pull/4245/files#diff-5db885ce8d97cd7d633bda7736e2db655fffb988a7b012e1a76007517d90af42R195)